src/saveload/saveload.h

```diff
@@ -594,7 +625,22 @@ struct SaveLoad {
 	size_t size;                    ///< the sizeof size.
 	SaveLoadAddrProc *address_proc; ///< callback proc the get the actual variable address in memory
 	size_t extra_data;              ///< extra data for the callback proc
-	SaveLoadHandler *handler;       ///< Custom handler for Save/Load procs.
+	std::shared_ptr<SaveLoadHandler> handler; ///< Custom handler for Save/Load procs.
+};
+
+/**
+ * SaveLoad information for backwards compatibility.
+ *
+ * At SLV_SETTINGS_NAME a new method of keeping track of fields in a savegame
+ * was added, where the order of fields is no longer important. For older
+ * savegames we still need to know the correct order. This struct is the glue
+ * to make that happen.
+ */
+struct SaveLoadCompat {
+	std::string name;             ///< Name of the field.
+	uint16 length;                ///< Length of the NULL field.
+	SaveLoadVersion version_from; ///< Save/load the variable starting from this savegame version.
+	SaveLoadVersion version_to;   ///< Save/load the variable until this savegame version.
```

Comment:
"until" is confusing combined with `IsSavegameVersionUntil` as the latter until includes the version, whereas here that version is explicitly not included (`version_to` is tested with `IsSavegameVersionBefore`). I know it's copied from `SaveLoad`, so that comment might need to be improved too.

Reply:
Lol, that is a bit ironic, as "until" is excluding (it means "before", after all). So that means `IsSavegameVersionUntil` is being the problem-child here :) I will address all this in a separate commit in this PR :)